js-api-doc/compile.d.ts

```diff
@@ -37,17 +37,121 @@ export interface CompileResult {
   sourceMap?: RawSourceMap;
 }
 
+/**
+ * The result of creating a synchronous compiler. Returned by
+ * {@link initCompiler}.
+ *
+ * @category Compile
+ */
+export class Compiler {
+  /**
+   * Throws an error if constructed directly, instead of via
+   * {@link initCompiler}.
+   */
+  private constructor();
```

Comment:
If we don't want to expose this with constructor, I think we can just define `Compiler` as an `interface` instead of a `class`, and then have an internal `class CompilerImpl implements Compiler` that is not exported so that it cannot be invoked by the user. `initCompiler` basically just call `new CompilerImpl()`.